### PR TITLE
Cumulus 1418

### PR DESCRIPTION
**Pull request by Jkovarik**
This PR is the cumulus-ecs-task portion of [CUMULUS-1418](https://bugs.earthdata.nasa.gov/browse/CUMULUS-1418).   This adds the ability of the cumulus-ecs-task package to bring in lambda layers (needed if the CMA is going to be a layer) as well as handle old-style CMA injection.    

There's a fair bit of refactoring to be done here outside these changes,  however that's out of scope of this ticket/PR.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -268,21 +317,19 @@ async function runTask(options) {
   assert(options && typeof options.lambdaInput === 'object', 'options.lambdaInput object is required');
   assert(options.taskDirectory && typeof options.taskDirectory === 'string', 'options.taskDirectory string is required');
   assert(options.workDirectory && typeof options.workDirectory === 'string', 'options.workDirectory string is required');
+  assert(!options.layersDirectory || typeof options.layersDirectory === 'string', 'options.layersDirectory should be a string');
```
Comment (Jkovarik) [inline index.js]: There's refactoring to be done here, however that's outside the scope of the current ticket to some degree.    I opted for consistency here.

File: package.json [hunk 2]
```diff
@@ -20,7 +20,7 @@
   },
   "license": "Apache-2.0",
   "dependencies": {
-    "aws-sdk": "^2.203.0",
+    "aws-sdk": "^2.503.0",
```
Comment (Jkovarik) [inline package.json]: Required for layer related API calls

File: index.js [hunk 3]
```diff
@@ -79,18 +82,38 @@ function downloadFile(url, destinationFilename) {
...
 * @param {string} arn - the arn of the lambda function
-* @param {strind} workDir - the dir to download the lambda function to
+* @param {string} workDir - the dir to download the lambda function to
+* @param {string} layersDir - the dir layers will be downloaded to
 * @returns {Promise<Object>} returns an object that includes `filepath`,
 * `moduleFileName`, `moduleFunctionName` arguments.
 * The `filepath` is the path to the zip file of the lambda function.
 * The `moduleFileName` is the filename of the node module.
 * The `moduleFunctionName` is the name of the exported function to call in the module.
+* The 'layerPaths' is an array of filepaths to downloaded layer zip files
```
Comment (markdboyd) [inline index.js]:
```suggestion
* The `layerPaths` is an array of filepaths to downloaded layer zip files
```

**Comment by Jkovarik** [comment]
@markdboyd  thanks for the suggestion commits.   I've added them, I think this is good to go.

**Pull request merged**